pkg/synchromanager/clustersynchro/resource_synchro.go

```diff
@@ -91,7 +93,45 @@ func newResourceSynchro(cluster string, syncResource schema.GroupVersionResource
 	return synchro
 }
 
-func (synchro *ResourceSynchro) Run(stopCh <-chan struct{}) {
+func (synchro *ResourceSynchro) Run(shutdown <-chan struct{}) {
+	go func() {
+		select {
+		case <-shutdown:
+			synchro.Close()
+		case <-synchro.closer:
+		}
+	}()
+
+	var waitGroup sync.WaitGroup
+	for i := 0; i < synchro.workerNum; i++ {
+		waitGroup.Add(1)
+
+		go wait.Until(func() {
+			defer waitGroup.Done()
+			synchro.processResources()
+		}, time.Second, synchro.closer)
+	}
+	waitGroup.Wait()
+
+	status := clusterv1alpha2.ClusterResourceSyncCondition{
+		Status:             clusterv1alpha2.SyncStatusStop,
+		LastTransitionTime: metav1.Now().Rfc3339Copy(),
+	}
+	synchro.status.Store(status)
```

Comment:
Is this the desired final state? The status maybe overwritten by synchro.status.Store operation in start method.